src/main/java/ratpack/pac4j/RatpackPac4j.java

```diff
@@ -510,39 +499,43 @@ public static Operation logout(Context ctx) {
   private static void initiateAuthentication(Context ctx, Class<? extends Client> clientType) {
     Request request = ctx.getRequest();
     Clients clients = ctx.get(Clients.class);
-    Client<?, ?> client = clients.findClient(clientType);
+    Client client = clients.findClient(clientType).get();
 
     RatpackWebContext.from(ctx, false).then(webContext -> {
       webContext.getSessionStore().set(webContext, Pac4jSessionKeys.REQUESTED_URL.getName(), request.getUri());
       try {
-        client.redirect(webContext);
+        Optional<RedirectionAction> redirect = client.getRedirectionAction(webContext, webContext.getSessionStore());
+        if (redirect.isPresent() && redirect.get() instanceof WithLocationAction) {
```

Comment:
Might be able to make this a bit cleaner using `filter` and `ifPresentOrElse`.

Reply:
That's Java 9 too unfortunately, but I can clean it up a bit with a filter and map to cast it.